comp/logs/integrations/impl/integrations.go

```diff
@@ -0,0 +1,38 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+// Package impl implements the integrations component interface
+package integrations
+
+import (
+	integrations "github.com/DataDog/datadog-agent/comp/logs/integrations/def"
+)
+
+type logsintegration struct {
+	logChan chan integrations.IntegrationLog
+}
+
+// NewComponent creates a new integrations component
+func NewComponent() integrations.Component {
+	return &logsintegration{
+		logChan: make(chan integrations.IntegrationLog),
+	}
+}
+
+// SendLog sends a log to any subscribers
+func (li *logsintegration) SendLog(log, integrationID string) {
+	integrationLog := &integrations.IntegrationLog{
```

Comment:
Thanks! In what cases would it be preferred to pass by reference rather than pass by value?

Reply:
*it depends* 
- It's good to try to pass value types over channels when possible because it removes the possibility of concurrent mutation since the data is always copied. 
- Sometimes our objects may be very large and it could impact performance to copy them every time you send them over a channel. In these cases it may be useful to send a pointer over the channel but you must also ensure the data is not mutated by the sending thread after it is received by the receiving thread. 

We generally try to design systems were data has some sense of ownership to prevent these kinds of issues. For example in the logs agent - a log message (and it's metadata) is passed between pipeline stages via channels. The log message can be mutated in each stage, but never mutated in multiple stages at once.